src/gui/systray.cpp

```diff
@@ -133,123 +126,225 @@ void Systray::setToolTip(const QString &tip)
+            trayIconCenter.y()
+        };
+    case TaskBarPosition::Top:
+        return {
+            trayIconCenter.x(),
+            screenRect.top() + taskbarRect.height() + 4
+        };
+    case TaskBarPosition::Right:
+        return {
+            screenRect.right() - taskbarRect.width() - 4,
+            trayIconCenter.y()
+        };
     }
-#endif
+    Q_UNREACHABLE();
 }
 
-bool Systray::syncIsPaused()
+QPoint Systray::computeWindowPosition(int width, int height) const
 {
-    return _syncIsPaused;
+    const auto referencePoint = computeWindowReferencePoint();
+
+    const auto taskbarScreenEdge = taskbarOrientation();
+    const auto screenRect = currentScreenRect();
+
+    const auto topLeft = [=]() {
+        switch(taskbarScreenEdge) {
+        case TaskBarPosition::Bottom:
+            return referencePoint - QPoint(width / 2, height);
+        case TaskBarPosition::Left:
+            return referencePoint;
+        case TaskBarPosition::Top:
+            return referencePoint - QPoint(width / 2, 0);
+        case TaskBarPosition::Right:
+            return referencePoint - QPoint(width, 0);
+        }
+        Q_UNREACHABLE();
+    }();
+    const auto bottomRight = topLeft + QPoint(width, height);
+    const auto windowRect = [=]() {
+        const auto rect = QRect(topLeft, bottomRight);
+        auto offset = QPoint();
+
+        if (rect.left() < screenRect.left()) {
+            offset.setX(screenRect.left() - rect.left() + 4);
+        } else if (rect.right() > screenRect.right()) {
+            offset.setX(screenRect.right() - rect.right() - 4);
+        }
+
+        if (rect.top() < screenRect.top()) {
+            offset.setY(screenRect.top() - rect.top() + 4);
+        } else if (rect.bottom() > screenRect.bottom()) {
+            offset.setY(screenRect.bottom() - rect.bottom() - 4);
+        }
+
+        return rect.translated(offset);
+    }();
+
+    qCDebug(lcSystray) << "taskbarScreenEdge:" << taskbarScreenEdge;
+    qCDebug(lcSystray) << "screenRect:" << screenRect;
+    qCDebug(lcSystray) << "windowRect (reference)" << QRect(topLeft, bottomRight);
+    qCDebug(lcSystray) << "windowRect (adjusted )" << windowRect;
+
+    return windowRect.topLeft();
 }
 
-void Systray::pauseResumeSync()
+/// Returns a QPoint that constitutes the center coordinate of the tray icon
+QPoint Systray::calcTrayIconCenter() const
 {
-    if (_syncIsPaused) {
-        _syncIsPaused = false;
-        emit resumeSync();
-    } else {
-        _syncIsPaused = true;
-        emit pauseSync();
-    }
+    // QSystemTrayIcon::geometry() is broken for ages on most Linux DEs (invalid geometry returned)
+    // thus we can use this only for Windows and macOS
+#if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
+    auto trayIconCenter = geometry().center();
+    return trayIconCenter;
+#else
+    // On Linux, fall back to mouse position (assuming tray icon is activated by mouse click)
+    return QCursor::pos();
```

Comment:
I think this is cool even if the tray icon is not activated by mouse click. :smiley: